plugins/woocommerce/src/Admin/Features/NewProductManagementExperience.php

```diff
@@ -54,6 +56,37 @@ public function maybe_show_disabled_notice() {
 		}
 	}
 
+	/**
+	 * Enqueue scripts needed for the product form block editor.
+	 */
+	public function enqueue_scripts() {
+		if ( ! PageController::is_admin_or_embed_page() ) {
+			return;
+		}
+		$post_type_object     = get_post_type_object( 'product' );
+		$block_editor_context = new WP_Block_Editor_Context( array( 'name' => 'core/edit-post' ) );
+
+		$editor_settings = array();
+		if ( ! empty( $post_type_object->template ) ) {
+			$editor_settings['template']     = $post_type_object->template;
+			$editor_settings['templateLock'] = ! empty( $post_type_object->template_lock ) ? $post_type_object->template_lock : false;
+		}
+
+		if ( wp_is_block_theme() && $editor_settings['supportsTemplateMode'] ) {
```

Comment:
We probably need to check if this property is set first.

<img width="738" alt="Screen Shot 2023-03-08 at 10 17 00 AM" src="https://user-images.githubusercontent.com/10561050/223797010-550cc77e-e9ce-49d4-862e-62528b893183.png">

Reply:
I actually decided to remove this condition, after re-reading what this logic does it seemed not overly relevant for our use case ( I think ): https://github.com/woocommerce/woocommerce/pull/37123/commits/e8a212f164dad401679393e0ab967b1c876e6dc6
Given both conditions are to see if we have a block theme and if the theme supports block templates (neither of which we actually care about as the product screen is not theme dependant ).